File: api/src/routes/results.js
import { Router } from 'express';
import { getResult, updateResult } from '../db/results.js';

const router = Router();

/**
 * @openapi
 * /api/results/{id}:
 *   get:
 *     summary: Get a single result
 *     tags: [Results]
 *     parameters:
 *       - in: path
 *         name: id
 *         required: true
 *         schema:
 *           type: string
 *           format: uuid
 *     responses:
 *       200:
 *         description: Result details
 *         content:
 *           application/json:
 *             schema:
 *               $ref: '#/components/schemas/Result'
 *       404:
 *         description: Result not found
 *
 *   patch:
 *     summary: Update a result
 *     description: Update fields like verified status or notes for a result.
 *     tags: [Results]
 *     parameters:
 *       - in: path
 *         name: id
 *         required: true
 *         schema:
 *           type: string
 *           format: uuid
 *     requestBody:
 *       required: true
 *       content:
 *         application/json:
 *           schema:
 *             type: object
 *             properties:
 *               verified:
 *                 type: boolean
 *               notes:
 *                 type: string
 *     responses:
 *       200:
 *         description: Updated result
 *       400:
 *         description: No valid fields to update
 *       404:
 *         description: Result not found
 */
router.get('/:id', (req, res) => {
  try {
    const result = getResult(req.params.id);
    if (!result) return res.status(404).json({ error: 'Result not found' });
    res.json(result);
  } catch (err) {
    console.error('GET /api/results/:id error:', err);
    res.status(500).json({ error: err.message });
  }
});

router.patch('/:id', (req, res) => {
  try {
    const existing = getResult(req.params.id);
    if (!existing) return res.status(404).json({ error: 'Result not found' });

    const allowed = ['verified', 'notes'];
    const updates = {};
    for (const key of allowed) {
      if (req.body[key] !== undefined) {
        updates[key] = req.body[key];
      }
    }

    if (Object.keys(updates).length === 0) {
      return res.status(400).json({ error: 'No valid fields to update' });
    }

    const updated = updateResult(req.params.id, updates);
    res.json(updated);
  } catch (err) {
    console.error('PATCH /api/results/:id error:', err);
    res.status(500).json({ error: err.message });
  }
});

export default router;

File: api-python/routes/results.py
from fastapi import APIRouter, HTTPException

from database import get_result, update_result
from models import ResultResponse, ResultUpdate

router = APIRouter(prefix='/api/results', tags=['Results'])


@router.get('/{result_id}', response_model=ResultResponse,
    summary='Get a single result',
    description='Returns details for a single business result by ID.')
async def get_result_endpoint(result_id: str = ...):
    result = get_result(result_id)
    if not result:
        raise HTTPException(404, 'Result not found')
    return result


@router.patch('/{result_id}', response_model=ResultResponse,
    summary='Update a result',
    description='Update verified status or notes for a business result.')
async def update_result_endpoint(result_id: str, body: ResultUpdate):
    existing = get_result(result_id)
    if not existing:
        raise HTTPException(404, 'Result not found')

    allowed = {'verified', 'notes'}
    updates = body.model_dump(exclude_none=True)
    updates = {k: v for k, v in updates.items() if k in allowed}

    if not updates:
        raise HTTPException(400, 'No valid fields to update')

    db_updates = {}
    if 'verified' in updates:
        db_updates['website_verified'] = 1 if updates['verified'] else 0
        db_updates['website_confirmed'] = 1 if updates['verified'] else 0
    if 'notes' in updates:
        db_updates['notes'] = updates['notes']

    updated = update_result(result_id, **db_updates)
    return updated
